app/models/mhv_account.rb

```diff
@@ -20,20 +22,31 @@ class MhvAccount < ActiveRecord::Base
 
   after_initialize :setup
 
+  # rubocop:disable Metrics/BlockLength
   aasm(:account_state) do
     state :unknown, initial: true
-    state :needs_terms_acceptance, :existing, :ineligible, :registered, :upgraded, :register_failed, :upgrade_failed
+    state :needs_ssn_resolution,
```

Comment:
Not a big deal, but you could clean up the duplication here since you have ALL_STATES .. you could splat everything except unknown like `*(ALL_STATES - [:unknown])` or `*ALL_STATES[1..-1]`